src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -47,34 +76,101 @@ const handsontable = new Handsontable(container, {
  * Pushes the output data into the textarea (used for form submission).
  */
 function updateDataDump() {
-    const spreadsheetData = handsontable.getData();
-    const dataPushToTextarea = getUpdatedHeaderString(handsontable.getColHeader());
+    const spreadsheetData = enrollHandsontable.getData();
+    const dataPushToTextarea = getUpdatedHeaderString(enrollHandsontable.getColHeader());
     const userDataRows = getUserDataRows(spreadsheetData);
     $('#enrollstudents').text(userDataRows === '' ?
             '' : dataPushToTextarea + userDataRows); // only pushes header string if userDataRows is not empty
 }
 
+/**
+ * Loads existing student data into the spreadsheet interface.
+ * @param studentsData
+ * @param $panelHeading
+ * @param panelCollapse
+ * @param displayIcon
+ * @param toggleChevron
+ */
+function loadExistingStudentsData(studentsData, $panelHeading, panelCollapse, displayIcon, toggleChevron) {
+    dataHandsontable.loadData(getExistingStudentsData(
+            studentsData, dataHandsontable.getColHeader()));
+    toggleExistingStudentsPanel($panelHeading, panelCollapse,
+            displayIcon, toggleChevron);
+    dataHandsontable.render(); // needed as the view is buggy after collapsing the panel
+}
+
+/**
+ * Gets list of student data through an AJAX request.
+ * @param $panelHeading
+ * @param panelCollapse
+ * @param displayIcon
+ * @param toggleChevron
+ */
+/*  eslint no-unused-expressions: [2, { allowTernary: true }]   */
+function getAjaxStudentList($panelHeading, panelCollapse, displayIcon, toggleChevron) {
+    const $spreadsheetForm = $('#student-data-spreadsheet-form');
+
+    $.ajax({
+        type: 'POST',
+        url: '/page/instructorCourseEnrollAjaxPage',
+        cache: false,
+        data: {
+            courseid: $spreadsheetForm.children(`input[name="${ParamsNames.COURSE_ID}"]`).val(),
+            user: $spreadsheetForm.children(`input[name="${ParamsNames.USER_ID}"]`).val(),
+        },
+        beforeSend() {
+            displayIcon.html('<img height="25" width="25" src="/images/ajax-preload.gif">');
+        },
+        error() {
+            displayErrorExecutingAjax(displayIcon);
+        },
+        success(data) {
+            (data.students.length === 0) ? displayNoExistingStudents(displayIcon) :
+            loadExistingStudentsData(data.students, $panelHeading, panelCollapse, displayIcon, toggleChevron);
+        },
+    });
+}
+
+/**
+ * Function to collapse "Existing students" panel and loads existing students' data when the user clicks on it.
+ */
+function collapseExistingStudentsPanel() {
```

Comment:
What if the user click again?

Reply:
It will perform a collapse hide. I'll update the function description to state that too.

Reply:
And the function name too, since it does both show and hide, not just hiding